modules/tinymce/src/core/main/ts/delete/TableDelete.ts

```diff
@@ -50,13 +50,28 @@ const handleEmptyBlock = (editor: Editor, startInTable: boolean, emptyBlock: Opt
   });
 };
 
-const setCursorInCell = (editor: Editor, cell: SugarElement<HTMLTableCellElement>) => {
+const deleteContentInsideCell = (cell: SugarElement<HTMLTableCellElement>, rng: Range, isFirstCellInSelection: boolean) => {
+  const insideTableRng = rng.cloneRange();
+
+  if (isFirstCellInSelection) {
+    insideTableRng.setStart(rng.startContainer, rng.startOffset);
+    insideTableRng.setEndAfter(cell.dom.lastChild);
+  } else {
+    insideTableRng.setStartBefore(cell.dom.firstChild);
+    insideTableRng.setEnd(rng.endContainer, rng.endOffset);
+  }
+
+  insideTableRng.deleteContents();
+};
+
+const collapseAndRestoreCellSelection = (editor: Editor) => {
   const selectedCells = TableCellSelection.getCellsFromEditor(editor);
-  editor.selection.setCursorLocation(cell.dom, 0);
+  editor.selection.collapse(true);
   // Restore the data-mce-selected attribute if multiple cells were selected, as if it was a cef element
   // then selection overrides would remove it as it was using an offscreen selection clone.
   if (selectedCells.length > 1) {
-    Attribute.set(cell, 'data-mce-selected', '1');
+    const firstCell = Arr.find(selectedCells, (cell) => Attribute.has(cell, 'data-mce-first-selected')).getOr(selectedCells[0]);
```

Comment:
I mentioned this yesterday in passing, but we can't use `data-mce-first-selected` in core, as that's a table plugin specific attribute.

- [ ] Refactor this so that we don't use `data-mce-first-selected` or `data-mce-last-selected`.

Reply:
We're probably going to have to use where the selection is after collapsing and then restore based on which cell the selection is in/on.